wildfly-getting-started-archetype/src/main/resources/archetype-resources/src/test/java/__defaultClassPrefix__ApplicationIT.java

```diff
@@ -0,0 +1,39 @@
+#set( $symbol_pound = '#' )
+#set( $symbol_dollar = '$' )
+#set( $symbol_escape = '\' )
+package ${package};
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URI;
+
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.Response;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Run integration tests against the server and the deployed application.
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class ${defaultClassPrefix}ApplicationIT {
+
+    @Test
+    public void testHelloEndpoint() {
+        try (Client client = ClientBuilder.newClient()) {
+            Response response = client
+                    .target(URI.create("http://localhost:8080/"))
```

Comment:
We should inject a `URL` or `URI` with the `@ArquillianResource` here.

Reply:
Please get ridden of arquillian IMHO, it's dead/zombie project and it is not openshift compatible. We are moving on from it on QS and same should be done on archetypes imho.

Reply:
Arquillian is the only solution we have to test CDI beans afaict.

Reply:
So we are going to advise users to go for something that won't be able to test on the cloud? We can use http, rest, etc for the test to interact with the app...

Reply:
how do you test CDI bean in Jakarta EE application otherwise?

Reply:
I dunno, not even looking into that honestly :-) Just think that before you decide how to implement your testing you should use ensure that the test can be run in all supported runtime scenarios. Arquillian is not just unsupported and not under development, it doesn't support such a critical priority named OpenShift. In the enhanced QS so far we have been relying only on http client to interact with the app but like I said there are other clients and even WFLY remote tools to use.